src/esp/physics/PhysicsManager.h

```diff
@@ -182,7 +182,7 @@ class PhysicsManager {
    * increments of @ref fixedTimeStep_.
    * @param dt The desired amount of time to advance the physical world.
    */
-  virtual void stepPhysics(double dt = 0.0);
+  virtual void stepPhysics(double dt);
```

Comment:
I understand why it complains (because an override could be defining a different default), but nevertheless I think defaults in overrideable APIs are useful. What I'm doing in Magnum is, roughly the following, based on the ages-old [Virtuality article by Herb Sutter](http://www.gotw.ca/publications/mill18.htm):

```cpp
// non-virtual, public
void stepPhysics(double dt = 0.0) {
    // do various error checking here, so the (re)implementations don't need to duplicate that
    doStepPhysics(dt);
}

// virtual with no default parameters, ideally protected or private
virtual void doStepPhysics(double dt);
```

Reply:
Thanks much for the pointer to this excellent article. Great read!

If we were to follow the approach outlined in the article, then defaults in overrideable APIs would not be needed since the public interface is not overrideable.

Since this is contentious, I will remove it. My immediate goal is to enable coding style verification and get a baseline setup.

Reply:
You could use an overload instead of an override in order to get default behavior which is defined in exactly one place.

Reply:
True, good idea -- just so we're on the same page, you mean a non-virtual overload with one parameter less that calls the virtual function with the default value, right? Like

```cpp
void stepPhysics() {
    stepPhysics(0.0);
}
```

However I have a feeling some static analyzers complain when you have overloaded virtuals.